flyway-core/src/main/java/org/flywaydb/core/internal/database/yugabytedb/YugabyteDBDatabaseType.java

```diff
@@ -0,0 +1,50 @@
+package org.flywaydb.core.internal.database.yugabytedb;
+
+import org.flywaydb.core.api.ResourceProvider;
+import org.flywaydb.core.api.configuration.Configuration;
+import org.flywaydb.core.internal.database.base.Database;
+import org.flywaydb.core.internal.database.postgresql.PostgreSQLDatabaseType;
+import org.flywaydb.core.internal.jdbc.JdbcConnectionFactory;
+import org.flywaydb.core.internal.jdbc.StatementInterceptor;
+import org.flywaydb.core.internal.parser.Parser;
+import org.flywaydb.core.internal.parser.ParsingContext;
+
+import java.sql.Connection;
+
+
+public class YugabyteDBDatabaseType extends PostgreSQLDatabaseType {
+    @Override
+    public String getName() {
+        return "YugabyteDB";
+    }
+
+    @Override
+    public boolean handlesJDBCUrl(String url) {
+        return url.startsWith("jdbc:postgresql:") || url.startsWith("jdbc:p6spy:postgresql:");
+    }
+
+    @Override
+    public int getPriority() {
```

Comment:
I remember you said Cockroach also uses a similar technique of bumping up their priority. Will this conflict in an environment where both Cockroach and YugabyteDBs are there?

Reply:
This method only determines which implementation will be evaluated first for a given database url.
But ultimately, the implementation is decided based on another method `handlesDatabaseProductNameAndVersion()`.